### PR TITLE
Fix not being able to reopen signals panel immediately after connecting a signal

### LINKED ISSUE #92996: Can't reopen signals panel immediately after connecting a signal
### Tested versions

- Reproduces in v4.3.beta1.official [a4f2ea91a]
- Doesn't reproduce in v4.3.dev6.official [89850d553]

### System information

Windows 10.0.22631 - Vulkan (Forward+) - dedicated AMD Radeon RX 7700 XT (Advanced Micro Devices, Inc.; 31.0.24027.1012) - AMD Ryzen 7 5700X3D 8-Core Processor (16 Threads)

### Issue description

After a signal is connected, the signals panel becomes empty. This I'm used to, but when I try to reopen it for the same node by clicking it in the scene tree, nothing happens. I can only reopen the signals panel for the same node after unselecting and reselecting it.

On a side note, why does the signals panel forget the node after connecting a signal? If it's not necessary, it'd be nice if it didn't.

### Steps to reproduce

- Select a node in the scene tree
- Open the signals panel
- Connect any signal
- Try reopen the same signals panel by clicking the same node in the scene tree

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by Hilderin**
Fixes #92996 

The issue was that `last_valid_node` was being set to null when `p_node` was null, causing the loss of the last valid node. After a signal connection, `restore_last_valid_node` is called, but `last_valid_node` no longer contained the node being edited.

### CODE REVIEW

**Comment by KoBeWi** [comment]
Something is wrong. The Node dock shows the connecting node, but the inspector does not.

https://github.com/godotengine/godot/assets/2223172/24c14a63-4e21-43bf-81cd-dbcf4d7ca96d

**Comment by Hilderin** [comment]
I changed a couple of things because it's was weird that the NodeDock and Inspector were not working the same way.

I changed the fix that was made from #84530 simply by add the line `NodeDock::get_singleton()->set_node(nullptr);` in `scene_tree_dock.cpp` when the node is deleted.
Also, I added a `previous_item` method in `EditorNode`  so the `_add_callback` could select back the last node.
That allowed me to remove everything managing the last valid node in NodeDock.

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -2304,6 +2304,12 @@ void EditorNode::push_item(Object *p_object, const String &p_property, bool p_in
 	_edit_current();
 }
 
+void EditorNode::previous_item() {
```
Comment (KoBeWi) [inline editor/editor_node.cpp:2307]:
```suggestion
void EditorNode::edit_previous_item() {
```
Make more sense I think.

File: editor/plugins/script_editor_plugin.cpp [hunk 2]
```diff
@@ -2856,6 +2856,8 @@ void ScriptEditor::_add_callback(Object *p_obj, const String &p_function, const
 
 		break;
 	}
+
+	EditorNode::get_singleton()->previous_item();
```
Comment (KoBeWi) [inline editor/plugins/script_editor_plugin.cpp:2860]: This might deserve a comment, as it makes an assumption about editing history.

**Review by KoBeWi — APPROVED** [review]

**Comment by KoBeWi** [comment]
Note that `restore_last_valid_node()` was added in #54071 to fix the same issue, but seems like it no longer works correctly for some reason.

File: editor/editor_node.cpp (hunk 1 above)
Comment (Hilderin) [reply editor/editor_node.cpp]: Done!

File: editor/plugins/script_editor_plugin.cpp (hunk 2 above)
Comment (Hilderin) [reply editor/plugins/script_editor_plugin.cpp]: Done!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**